cmake/EthCompilerSettings.cmake

```diff
@@ -41,75 +50,23 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 		endif ()
 	endif ()
 
-	# Additional GCC-specific compiler settings.
+	# Check GCC compiler version.
 	if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
-
-		# Check that we've got GCC 4.7 or newer.
-		execute_process(
-			COMMAND ${CMAKE_CXX_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
-		if (NOT (GCC_VERSION VERSION_GREATER 4.7 OR GCC_VERSION VERSION_EQUAL 4.7))
-			message(FATAL_ERROR "${PROJECT_NAME} requires g++ 4.7 or greater.")
-		endif ()
-
-		# Strong stack protection was only added in GCC 4.9.
-		# Use it if we have the option to do so.
-		# See https://lwn.net/Articles/584225/
-		if (GCC_VERSION VERSION_GREATER 4.9 OR GCC_VERSION VERSION_EQUAL 4.9)
-			add_compile_options(-fstack-protector-strong)
-			add_compile_options(-fstack-protector)
+		if("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS 4.8)
+			message(FATAL_ERROR "This compiler ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} is not supported. GCC 4.8 or newer is required.")
 		endif()
 
-	# Additional Clang-specific compiler settings.
-	elseif ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-
-		# Stop if buggy clang compiler detected.
-		if("${CMAKE_CXX_COMPILER_ID}" MATCHES AppleClang)
-			if("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS 8.4)
-				message(FATAL_ERROR "This compiler ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} is not able to compile required libff. Install clang 4+ from Homebrew.")
-			endif()
-		endif()
-
-
-		add_compile_options(-fstack-protector)
-
-		# Enable strong stack protection only on Mac and only for OS X Yosemite
-		# or newer (AppleClang 7.0+).  We should be able to re-enable this setting
-		# on non-Apple Clang as well, if we can work out what expression to use for
-		# the version detection.
-
-		# The fact that the version-reporting for AppleClang loses the original
-		# Clang versioning is rather annoying.  Ideally we could just have
-		# a single cross-platform "if version >= 3.4.1" check.
-		#
-		# There is debug text in the else clause below, to help us work out what
-		# such an expression should be, if we can get this running on a Trusty box
-		# with Clang.  Greg Colvin previously replicated the issue there too.
-		#
-		# See https://github.com/ethereum/webthree-umbrella/issues/594
-
-		if (APPLE)
-			if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0)
-				add_compile_options(-fstack-protector-strong)
-			endif()
-		else()
-			message(WARNING "CMAKE_CXX_COMPILER_VERSION = ${CMAKE_CXX_COMPILER_VERSION}")
-		endif()
-
-		# Some Linux-specific Clang settings.  We don't want these for OS X.
-		if ("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
-
-			# Tell Boost that we're using Clang's libc++.   Not sure exactly why we need to do.
-			add_definitions(-DBOOST_ASIO_HAS_CLANG_LIBCXX)
-
-			# Use fancy colors in the compiler diagnostics
-			add_compile_options(-fcolor-diagnostics)
```

Comment:
Color output and `BOOST_ASIO_HAS_CLANG_LIBCXX` is removed for the linux case.

Reply:
None of them are relevant any more.
1. We actually are using libstdc++ not libc++ on Linux.
2. Clang is using colors in terminal by default. That was for some old version.